### PR TITLE
feat: lsp.js!

### PR DESCRIPTION
**Pull request by ajbt200128**
PR checklist:

- [X] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [ ] Tests included or PR comment includes a reproducible test plan
- [X] Ran tests locally (VSCode tests cannot run in CI)
- [X] Documentation is up-to-date
- [X] A changelog entry was for any user-facing change
- [X] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Comment by ajbt200128** [comment]
Test plan:

merge this PR and add in lsp.js into https://github.com/semgrep/semgrep-vscode/pull/77

File: download-lspjs.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+mkdir -p lspjs/dist
```
Comment (tpetr) [inline download-lspjs.sh:1]: can you add a shebang to this file?

File: download-lspjs.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+for var in "$@"
+do
+    curl https://static.semgrep.dev/static/turbo/$(cat ./semgrep-version)/language_server/dist/$var -o ./lspjs/dist/$var
```
Comment (tpetr) [inline download-lspjs.sh:4]: what happens if a curl call fails? I think we might need a `set -e`

File: .vscode/launch.json [hunk 3]
```diff
@@ -10,6 +10,17 @@
...
+      "type": "node",
+      "request": "attach",
+      "name": "Attach to Server",
+      "port": 6009,
+      "restart": true,
+      "outFiles": [
+        "${workspaceRoot}/lspjs/out/**/*.js",
+        "${workspaceRoot}/lspjs/dist/*.js"
+      ]
+    },
```
Comment (tpetr) [inline .vscode/launch.json:23]: what does this do?

File: download-lspjs.sh (hunk 1 above)
Comment (ajbt200128) [reply download-lspjs.sh:1]: yup

File: download-lspjs.sh (hunk 2 above)
Comment (ajbt200128) [reply download-lspjs.sh:4]: Good point, will add that

File: .vscode/launch.json (hunk 3 above)
Comment (ajbt200128) [reply .vscode/launch.json:23]: This attaches the debugger to lsp.js

**Pull request merged**